src/core/border_router/routing_manager.cpp

```diff
@@ -340,9 +340,10 @@ void RoutingManager::RecvIcmp6Message(uint32_t            aInfraIfIndex,
 
     VerifyOrExit(IsInitialized() && mIsRunning, error = kErrorDrop);
     VerifyOrExit(aInfraIfIndex == mInfraIfIndex, error = kErrorDrop);
-    VerifyOrExit(aBuffer != nullptr && aBufferLength >= sizeof(*icmp6Header), error = kErrorParse);
+    VerifyOrExit(aBuffer != nullptr, error = kErrorInvalidArgs);
```

Comment:
@rretanubun I do not think  the value of `icmp6Header` pointer matters in this situation.
I think `sizeof` only cares about the type of its expersion. So this is equivalent to `sizeof(Ip6::Icmp::Header)`.

_From [sizeof operator (cppreference)](https://en.cppreference.com/w/cpp/language/sizeof):
When applied to an expression, sizeof does not evaluate the expression, and even if the expression designates a polymorphic object, the result is the size of the static type of the expression_

That said, it is possible that some code checker tools do not like the fact that `*icmp6Header` is being used without variable being set first. I suggest to use direct type in this case, i.e.
```
VerifyOrExit(aBuffer != nullptr && aBufferLength >= sizeof(Ip6::Icmp::Header), error = kErrorParse);
```

Reply:
Good point! Thanks for the explanation. Updated as suggested.